### ISSUE TITLE
[Improvement] Silent Alarms

### ISSUE THREAD

**Issue by chriske64** [MilosKozak/AndroidAPS#1304]
I'm using the RC6.1 with a combo pump. Sometimes I get alarms (eg. Pump unreachable, or pump history changed) during meetings. It is very unpleasant because it attracts all the attention and I always need to explain myself.
To overcome this problem I have put my phone into 'total silence' mode which supresses any sounds. But then I forgot to switch it off again and missed my waking up alarm. 

I would like to have a feature to completely turn off all sounds made by the app. Needless to say I think it should be not a default setting but I would really appreciate such a setting.

Thanks!

**Comment by triplem** [comment]
Hm, you state that you forget to turn on the alarms again. This happens to me as well, and I really have it especially for AAPS. Furthermore some alarms, at least for me, are important, even while sitting in a meeting with eg a customer (urgent low). I guess we should allow to customize urgent alarms and "normal" alarms differently. What do you think?

**Comment by chriske64** [comment]
Personally, I have muted the low/high alarms anyway. I can feel it etc.
I think you raise a good point but for me the most important part is that I can have a setting where the app does not make any sounds.

**Comment by andyrozman** [comment]
Yes silent mode is something that I would really need... Sometimes you just don't want to hear the alarms (important meetings, etc). 

Solution would actually be pretty simple... If user clicks on "Silent mode" all current alarm setting would be saved and all alarms turned off, and when user would select "Normal mode" again, all previous alarm settings would be restored.

**Comment by jroth-haj** [comment]
So far, as I know, this alarms are played as media. I also use dnd mode. The alarms are still coming up.
Because of that I have the media volume at 0.
When I hear music, I must raise up this.
When I forgot to raise down it after listening to music, I get loud alarms in urgent meetings. :-(

Why the app can't use system alarms?
The alarms also should be classified.

**Comment by lee-b** [comment]
I'm willing to (going to, if only for personal use) implement this "silent mode" (or something equivalent).  For one thing, the alarms were so distracting on a motorbike lately that it was dangerous.  I also have the work problem, like others mentioned above.

I'm new to android development, java development, and this codebase, but not dev more generally.  Any tips on where to start this?

I'm thinking it would be nice to implement this through the same notificationbuilder that other parts of the codebase (seem to) use for alerts, which support vibration instead of just sounds.  Don't really get why these particular alerts just play sounds directly.

However, as a start, I might just add a "silent mode" config option that overrides these particular alerts.

**Comment by lee-b** [comment]
For those of you getting the "pump unreachable" error frequently: do you often stream music over bluetooth (like I do)?  If so, do you find the audio choppy?

I'm **beginning** to suspect that the problem with the connectivity is bluetooth timeslicing.  **If** so, we **might** be able to solve the error, instead of fixing the alarms (albeit, the alarms are still too noisy).

**Comment by anbraten** [comment]
> Why the app can't use system alarms?
> The alarms also should be classified.

I have this alarms on calls as well. This is just a simple no go.
IMHO we have to should change the alarms to be alarms and not media sounds.

Is there a special reason to use a MediaPlayer instead of the RingtoneManager / NotificationManager?
If not, I would open a PR to replace the current MediaPlayer ([AlarmSoundService.java](https://github.com/MilosKozak/AndroidAPS/blob/master/app/src/main/java/info/nightscout/androidaps/services/AlarmSoundService.java#L43-L71
)) with something like the following:

```
try {
    Uri notification = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM);
    Ringtone r = RingtoneManager.getRingtone(getApplicationContext(), notification);
    r.play();
} catch (Exception e) {
    e.printStackTrace();
}
```
